presto-main/src/main/java/com/facebook/presto/sql/planner/RelationPlanner.java

```diff
@@ -650,7 +661,17 @@ protected RelationPlan visitUnnest(Unnest node, Void context)
             Symbol inputSymbol = symbolAllocator.newSymbol(rewritten, type);
             argumentSymbols.add(inputSymbol);
             if (type instanceof ArrayType) {
-                unnestSymbols.put(inputSymbol, ImmutableList.of(unnestedSymbolsIterator.next()));
+                Type elementType = ((ArrayType) type).getElementType();
```

Comment:
line 663 - 680 is same as line 562 - 580. And they are both doing unnest.

Would it make sense to refactor them? @martint  @kokosing

Reply:
I don't quite like how unnest is done through out the code base. the
```
if (type instanceof ArrayType)
...
else if (type instanceof MapType)
...
```
is everywhere. If I were to do a refactoring, I'd prefer something doesn't require that. But I didn't see an easy way to do so.

Reply:
Please extract the outer if as a static method.

I'm not sure what part of the following snippet you don't like 

```
if (type instanceof ArrayType)
...
else if (type instanceof MapType)
...
```

You could have done `switch(type.getTypeSignature().getBase()`. But that makes little difference.